test/type.query.test.js

```diff
@@ -207,6 +223,69 @@ describe('type: query', () => {
             return;
         });
     });
+    describe('FixKeys ================', () => {
+        beforeEach(() => {
+            testUtils.mockSetup(true);
+        });
+        it('Should not fixKeys and deploy', async () => {
+            // WHEN
+            await handler.fixKeys('testInstance/testBU', 'query', ['testExisting_query']);
+            // THEN
+            assert.equal(process.exitCode, false, 'fixKeys should not have thrown an error');
+            // get results from cache
+            const result = cache.getCache();
+            assert.equal(
+                result.query ? Object.keys(result.query).length : 0,
+                1,
+                'one query expected'
+            );
+            // check number of API calls
+            assert.equal(
+                testUtils.getAPIHistoryLength(),
+                7,
+                'Unexpected number of requests made. Run testUtils.logAPIHistoryDebug() to see the requests'
+            );
+            return;
+        });
+        it('Should fixKeys and deploy', async () => {
+            const properties = await config.getProperties();
+            const deployDir = properties.directories.deploy;
+            // WHEN
+            const resultFixKeys = await handler.fixKeys('testInstance/testBU', 'query', [
+                'testExisting_query_fixKeys',
+                'testExisting_query',
+            ]);
+            assert.equal(
+                resultFixKeys.length,
+                1,
+                'returned number of keys does not correspond to number of expected fixed keys'
+            );
+            assert.equal(
+                resultFixKeys[0],
+                'testExisting_query_fixedKeys',
+                'returned keys do not correspond to expected fixed keys'
+            );
+            // THEN
+            assert.equal(process.exitCode, false, 'fixKeys should not have thrown an error');
+            // confirm updated item
+            assert.deepEqual(
+                await testUtils.getActualJson('testExisting_query_fixedKeys', 'query'),
+                await testUtils.getExpectedJson('9999999', 'query', 'patch_fixKeys'),
+                'returned metadata was not equal expected for update query'
+            );
+            expect(
+                file(testUtils.getActualFile('testExisting_query_fixedKeys', 'query', 'sql'))
+            ).to.equal(file(testUtils.getExpectedFile('9999999', 'query', 'patch_fixKeys', 'sql')));
+            // check number of API calls
+            assert.equal(
+                testUtils.getAPIHistoryLength(),
+                16,
+                'Unexpected number of requests made. Run testUtils.logAPIHistoryDebug() to see the requests'
+            );
+            properties.directories.deploy = deployDir;
```

Comment:
@JoernBerkefeld 
I set deploy directory for fixKeys to retrieve dir, it causes problems in deploy script test. Is there anyway to reset execution context for when fixKeys test is done? This line is a workaround to avoid tests failing (see line 1078 in index.js)

Reply:
why dont you use the option fromRetrieve before running fixKeys()?


add this before your handler.fixKeys(...) call in your test class: 

```js
handler.setOptions({fromRetrieve:true});
```

Reply:
> ```js
> handler.setOptions({fromRetrieve:true});
> ```

The problem is in re-setting the deploy directory. I deploy from retrieve directory so I set deploy directory to retrieve. In tests before I attempt to deploy a script in test.script.test I need to set deploy directory back to deploy because tests fail. 
For example if I were to use Deployer.deploy() instead of deployBU in fixKeys() it will set deploy directory to retrieve, but in tests I still need to set it back to deploy/. 
![image](https://github.com/Accenture/sfmc-devtools/assets/79999614/9da068a4-74e2-44b3-b82a-60b1f20e959d)

Reply:
hi, actually my advice was only half correct:

using fromRetrieve:true IS the way to go, but only if you use the main Developer.develop method.
I removed that parameter that you screenshotted to make it consistent with similar options. Now, the methods look for Util.OPTIONS.fromRetrieve instead.

with that said, using handler.... is obviously wrong because that would only work for the test. instead, you'd have to manually set the option in fixKeys in the same way: `this.setOptions({fromRetrieve:true});`


if you need to go with deployBU() instead, then you'll have to adapt the hack that I added to your method:


before:
```js
        const deployDirBak = properties.directories.deploy;
        if (Util.OPTIONS.fromRetrieve) {
            properties.directories.deploy = properties.directories.retrieve;
        }
```

and after:
```js
        if (Util.OPTIONS.fromRetrieve) {
            properties.directories.deploy = deployDirBak;
        }
```

Reply:
PS: please merge develop again ;-)